docs/using-connector.md

```diff
@@ -533,6 +534,160 @@ conf.set("spark.riakts.write.bulk-size", "500")
+
+###Spark Streaming KV Buckets Example
+
+Now that we are set up, lets look at the KV bucket example [here](../examples/src/main/scala/com/basho/riak/spark/examples/streaming/StreamingKVExample.scala). 
+
+In the first chunk of code in the main method, we are just setting up our local spark streaming context and setting the name for the KV bucket to `test-data`:
+
+```scala
+val sparkConf = new SparkConf(true)
+      .setAppName("Simple Spark Streaming to Riak KV Demo")
+setSparkOpt(sparkConf, "spark.master", "local")
+setSparkOpt(sparkConf, "spark.riak.connection.host", "127.0.0.1:8087")
+
+val sc = new SparkContext(sparkConf)
+val streamCtx = new StreamingContext(sc, Durations.seconds(15))
+val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+import sqlContext.implicits._
+val namespace = new Namespace("test-data")
+```
+
+Next we are setting up kafka properties:
+
+```scala
+val kafkaProps = Map[String, String](
+      "metadata.broker.list" -> "127.0.0.1:9092",
+      "client.id" -> UUID.randomUUID().toString
+    )
+```
+
+Then, we are using `KafkaUtils` to create a stream from the kafka topic `streaming` into our KV bucket `test-data`:
+
+```scala
+    KafkaUtils
+      .createDirectStream[String, String, StringDecoder, StringDecoder](streamCtx, kafkaProps, Set[String]("streaming"))
+      .foreachRDD { rdd =>
+        val rows = sqlContext.read.json(rdd.values).map {
+          line => val obj = RiakObjectConversionUtil.to(line)
+            obj.setContentType("application/json")
+            obj
+        }.saveToRiak(namespace)
+      }
+```
+ And finally, we are starting the stream:
+ 
+ ```scala
+ streamCtx.start()
+ streamCtx.awaitTermination()
+ ```
+
+ Now that we understand the code, we can run the `StreamingKVExample.scala` example with:
+ 
+ ```
+ /path/to/spark-riak-connector-examples/bin/run-example streaming.StreamingKVExample
+ ```
+ 
+ This wil start a stream from the Kafka topic `streaming` into the KV bucket `test-data` that we just created. This stream will run until terminated. Whenever a message is produced for Kafka topic `streaming`, the spark streaming context that the example creates will automatically stream the message from the topic into the KV bukcet. To see this in action, we need to send a message to the kafka topic `streaming` with the kafka console producer script:
+ 
+ ```
+ /path/to/bin/kafka-console-producer.sh --broker-list localhost:9092 --topic streaming
+ ```
+ 
+ This will read a message from standard in and pass it to the topic. As an example put the following into standard in:
```

Comment:
Please add more detail about how to pass the JSON message to the console producer